src/coreclr/vm/stackwalk.cpp

```diff
@@ -2844,11 +2841,11 @@ void StackFrameIterator::ProcessCurrentFrame(void)
 #ifdef FEATURE_INTERPRETER
         if (!m_crawl.isFrameless)
         {
+            PREGDISPLAY pRD = m_crawl.GetRegisterSet();
+
             if (m_crawl.pFrame->GetFrameIdentifier() == FrameIdentifier::InterpreterFrame)
             {
-                PREGDISPLAY pRD = m_crawl.GetRegisterSet();
-
-                if (!m_walkingInterpreterFrames)
+                if (GetIP(pRD->pCurrentContext) != (TADDR)InterpreterFrame::DummyCallerIP)
```

Comment:
GetIP is being cast to TADDR below. Should it be cast to TADDR here as well? Or is that unnecessary?
It looks like GetIP returns PCODE. I assume the thumb bit can never be set because this is an interpreter frame, so comparing this interpreter PCODE with a TADDR is fine?

Reply:
I'll change the cast to PCODE, it makes sense.